app/admin/top-issues/components/TopIssueDisplay.js

```diff
@@ -0,0 +1,107 @@
+import React, { useState } from 'react';
+import { SVGIconChooser } from '@shared/buttons/SVGIconChooser';
+import TextField from '@shared/inputs/TextField';
+import { ButtonGroup } from '@mui/material';
+import Button from '@mui/material/Button';
+import { FaCheck, FaEdit } from 'react-icons/fa';
+import { FaXmark } from 'react-icons/fa6';
+import Image from 'next/image';
+import IconButton from '@mui/material/IconButton';
+import { updateTopIssue } from './TopIssuesList';
+
+export const TopIssueDisplay = ({
+  issue,
+  topIssues = [],
+  setTopIssues = () => {
+  },
+}) => {
+  const [editTopIssueId, setEditTopIssueId] = useState(null);
+  const [editTopIssueName, setEditTopIssueName] = useState(null);
+  const [editTopIssueIcon, setEditTopIssueIcon] = useState(null);
+
+  const handleIssueEdit = issue => () => {
+    setEditTopIssueId(issue.id);
+    setEditTopIssueName(issue.name);
+    setEditTopIssueIcon(issue.icon);
+  };
+
+  const handleClearIssueEdit = () => {
+    setEditTopIssueId(null);
+    setEditTopIssueName(null);
+    setEditTopIssueIcon(null);
+  };
+
+  const handleTopIssueUpdate = (issue) => async () => {
+    const updateIssue = {
+      ...issue,
+      id: editTopIssueId,
+      name: editTopIssueName,
+      icon: editTopIssueIcon,
+    };
+    const { icon } = await updateTopIssue({
+      ...issue,
+      ...updateIssue,
+    });
+    const issueIndx = topIssues.findIndex(({ id }) => id === issue.id);
+    setTopIssues(
+      [
+        ...topIssues.slice(0, issueIndx),
+        {
+          ...updateIssue,
+          ...(icon ? { icon } : {}),
+        },
+        ...topIssues.slice(issueIndx + 1),
+      ],
+    );
```

Comment:
this is not readable. Break it to multiple variable with meaningful names so we know what is happening here, and please refrain from using the spread operator so much